util/mod_translation_updater.py

```diff
@@ -272,9 +285,17 @@ def read_lua_file_strings(lua_file):
 	with open(lua_file, encoding='utf-8') as text_file:
 		text = text_file.read()
 
+		strings = []
+
+		for s in pattern_lua_quoted_single.findall(text):
+			strings.append(s[1])
```

Comment:
You can avoid the odd `s[1]` if you don't use a backreference and capture, but instead construct the regex as `'((?:\\.|[^\\']))*'|"((?:\\.|[^\\"]))*"`.

Reply:
That would result in even more capture groups (and would need checking each group's existence).

```python
import re
pattern = re.compile(r'(a)|(b)|(c)', re.DOTALL)
pattern.findall('abc')
# [('a', '', ''), ('', 'b', ''), ('', '', 'c')]
```

... or maybe we can just concatenate all since it is always one or the other.

```python
[''.join(x) for x in pattern.findall('abc')]
# ['a', 'b', 'c']
```

Reply:
Why write `(a)|(b)|(c)` instead of `((?:a)|(?:b)|(?:c))` (where `(?:x)` can be just `x` if precedency allows it)

Reply:
I was thinking about using it for `"..."`, `'...'`, and `[[...]]` which we also want to remove the delimiters.

```python
import re

pattern = re.compile(r'"((?:\\.|[^\"])*)"|\'((?:\\.|[^\\\'])*)\'|\[\[(.*?)\]\]')
sample_string = '"ABC\'DEF\\"", \'GHI"JKL\\\'\', [[MNOPQR]]'

pattern.findall(sample_string)
# [('ABC\'DEF\\"', '', ''), ('', 'GHI"JKL\\\'', ''), ('', '', 'MNOPQR')]

[''.join(x) for x in pattern.findall(sample_string)]
# ['ABC\'DEF\\"', 'GHI"JKL\\\'', 'MNOPQR']
```

The alternative might be trimming delimiters from the captured text.